src/main/java/bio/terra/app/controller/DatasetsApiController.java

```diff
@@ -281,6 +282,18 @@ public ResponseEntity<JobModel> deleteFile(
     return jobToResponse(jobService.retrieveJob(jobId, userReq));
   }
 
+  @Override
+  public ResponseEntity<List<FileModel>> listFiles(
+      @PathVariable("id") UUID id,
+      @Valid @RequestParam(value = "offset", required = false, defaultValue = "0") Integer offset,
+      @Valid @RequestParam(value = "limit", required = false, defaultValue = "10") Integer limit) {
+    AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    iamService.verifyAuthorization(
+        userRequest, IamResourceType.DATASET, id.toString(), IamAction.READ_DATASET);
+    List<FileModel> results = fileService.listDatasetFiles(id.toString(), offset, limit);
+    return new ResponseEntity<>(results, HttpStatus.OK);
```

Comment:
It's a minor thing but using the `.ok()` form is a little shorter
```suggestion
    return ResponseEntity.ok(results);
```